src/main/scala/viper/silver/parser/Resolver.scala

```diff
@@ -850,13 +849,17 @@ case class NameAnalyser() {
 
     val scopeStack = mutable.Stack[PScope]()
 
+    var globalDeclarationsFound = false
```

Comment:
Would it make sense to have two versions of `nodeDownNameCollectorVisitor` instead of this flag?